cmd/datasource-syncer/main.go

```diff
@@ -222,3 +238,43 @@ func buildUpdateDataSourceRequest(dataSource grafana.DataSource, token string) (
 	dataSource.SecureJSONData[authHeaderValue] = fmt.Sprintf("Bearer %s", token)
 	return &dataSource, nil
 }
+
+func getTLSClient(logger log.Logger, certFile, keyFile, caFile string, insecureSkipVerify bool) (*http.Client, error) {
+	if (certFile != "" || keyFile != "") && (certFile == "" || keyFile == "") {
+		return nil, fmt.Errorf("--tls-cert and tls-key must both be set or unset")
+	}
+
+	if certFile == "" && keyFile == "" && caFile == "" && !insecureSkipVerify {
+		return http.DefaultClient, nil
+	}
+
+	var client *http.Client
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: insecureSkipVerify,
+	}
+
+	if certFile != "" && keyFile != "" {
+		cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load server cert and key: %w", err)
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if caFile != "" {
+		caCert, err := os.ReadFile(caFile)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read ca cert: %w", err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		tlsConfig.RootCAs = caCertPool
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	client = &http.Client{Transport: transport}
```

Comment:
suggestion: you should use the default client and update the transport.

Reply:
breaks code.